File: agent-driven-architecture-main/examples/kotlin/block/console/src/main/kotlin/adr/blocks/console/ViewState.kt
// ── blocks/console/view-state — EPHEMERAL, and the only thing that is (4.6) ─
// Hover, scroll offset, which panel is expanded in THIS browser tab, unsubmitted
// text. None of it enters a tool, folds, or signs — losing it on a re-fold changes
// nothing the system believes and nothing the artifact contains.
//
// Gate check C12 keeps it that way: this file may be imported only by
// blocks/console/project, where ephemeral state joins the pre-decided ViewModel at
// the very last moment. `fold`, `slice`, `contract` and `register` cannot see it.
//
// 6.8's volume objection, answered: the axis is DECISION vs EPHEMERAL, not UI vs
// domain. A deliberate repositioning IS the "discrete, auditable, low-frequency
// action" 5.4 puts on the bus; a scroll offset is not.

package adr.blocks.console

import adr.spine.pure.TicketId

data class ViewState(
    val hover: TicketId? = null,
    val scrollOffset: Int = 0,
    val draft: String = "",
) {
    // No companion: a companion member has no instance, which is the same defect as a
    // top-level function. The EMPTY slice is now what the primary constructor builds
    // when told nothing — `ViewState()` — so the shape carries its own starting value and
    // nothing extra has to exist to hand it over.
}

File: agent-driven-architecture-main/examples/typescript/src/blocks/console/view-state.ts
// ── blocks/console/view-state — EPHEMERAL, and the ONLY exception (4.6) ────
// Hover, scroll offset, which panel is mid-drag, unsubmitted text. This NEVER
// enters a tool, NEVER folds, NEVER signs, and is never part of State.
//
// The test is 4.6's, verbatim: if losing the field on a re-fold would change
// what the system believes or what the artifact contains, it is truth — fold
// it. Nothing here passes that test. A scroll offset is not a decision; a
// deliberate repositioning is, and that one is a Command (see `contract.ts`).
//
// This file imports NOTHING and is imported only by `project.ts` (check C12),
// so there is no path by which it could reach the fold even by accident.

export interface ConsoleViewState {
  readonly hoveredTicket: string | null;
  readonly scrollOffset: number;
  readonly draft: string;
}

export const initialViewState: ConsoleViewState = {
  hoveredTicket: null,
  scrollOffset: 0,
  draft: "",
};
